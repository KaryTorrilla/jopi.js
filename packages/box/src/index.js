import styled from 'styled-components'
import {
  space,
  color,
  layout,
  flexbox,
  border,
  typography,
  position,
  variant,
} from 'styled-system'
import css from '@styled-system/css'

const sx = props => css(props.sx)(props.theme)
const base = props => css(props.__css)(props.theme)
const theme = ({ tx = 'variants' }) =>
  variant({
    scale: tx,
  })

export const Box = styled.div(
  {
    boxSizing: 'border-box',
    margin: 0,
    minWidth: 0,
  },
  base,
  theme,
  sx,
  space,
  color,
  layout,
  flexbox,
  typography,
  border,
  position
)

export const Flex = styled(Box)({
  display: 'flex',
})
