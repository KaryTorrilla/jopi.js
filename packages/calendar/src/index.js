import React from 'react'
import FullCalendar from '@fullcalendar/react' // must go before plugins
import dayGridPlugin from '@fullcalendar/daygrid' // a plugin!
import '@oneloop/theme/styles/globals.css'
import '../styles/styles.css'

export const Calendar = () => {
  /*   const calendarOptions = {
      events: [],
      plugins: [timeGridPlugin],
      initialView: 'timeGridDay',
      nowIndicator: true,
      headerToolbar: {
        start: 'prev',
        center: 'title',
        end: 'next',
      },
      editable: true,
      eventResizableFromStart: true,
      dragScroll: true,
      dayMaxEvents: 2,
      navLinks: true,
    } */

  return (
    <div className='calendarContainer'>
      <FullCalendar
        plugins={[dayGridPlugin]}
        initialView="dayGridMonth"
      />
    </div>
  )
}
