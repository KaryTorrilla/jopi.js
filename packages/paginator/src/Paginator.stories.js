import React, { useState } from 'react'
import { Paginator } from '.'
import { Text } from '@oneloop/text'

export default {
  component: Paginator,
  title: 'Paginator',
}

export const normal = () =>
  React.createElement(() => {
    const [actualPage, setActualPage] = useState(20)
    return (
      <React.Fragment>
        <Text variant="subtitle" m="30px">
          Mostrando p√°gina: {actualPage}
        </Text>
        <Paginator
          currentPage={actualPage}
          total={1500}
          perPage={25}
          changePageNumber={setActualPage}
        />
      </React.Fragment>
    )
  })
