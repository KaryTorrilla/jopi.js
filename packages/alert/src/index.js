import React from 'react'
import { Flex } from '@oneloop/box'
import { Check, Exclamation, Times } from '@oneloop/icons'
import { Text } from '@oneloop/text'
import { Button } from '@oneloop/button'

export const Alert = ({ type = 'success', text, ...props }) => (
  <Flex
    variant="alert"
    {...props}
    __css={{
      width: '100%',
      bg: 'white',
      borderRadius: 'circle',
      boxShadow: 'active',
      justifyContent: 'space-between',
      p: 1,
    }}
  >
    {type === 'success' && (
      <Check
        bg="success"
        color="white"
        borderRadius="circle"
        size={40}
        p={3}
        mr={5}
      />
    )}
    {type === 'warning' && (
      <Exclamation
        bg="warning"
        color="white"
        borderRadius="circle"
        size={40}
        p={3}
        mr={5}
      />
    )}
    {type === 'danger' && (
      <Times
        bg="danger"
        color="white"
        borderRadius="circle"
        size={40}
        p={3}
        mr={5}
      />
    )}
    <Text variant="body.0" display="flex" alignItems="center" flexGrow={1}>
      {text}
    </Text>
    <Button
      size={32}
      bg="#F6F2F2"
      p={3}
      display="inline-flex"
      alignSelf="center"
      mr={2}
    >
      <Times color="#716F6F" m="auto" />
    </Button>
  </Flex>
)
