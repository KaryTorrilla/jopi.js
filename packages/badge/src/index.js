import React from 'react'
import { Box } from '@oneloop/box'
import '@oneloop/fonts'

export const Badge = ({ isMedium, hasIcon, isNotButton, ...props }) => (
  <Box
    as={ isNotButton ? "" : "button" }
    tx="buttons"
    variant="primary"
    {...props}
    __css={{
      lineHeight: '14px',
      height: isMedium ? '22px' : '16px',
      display: 'inline-block',
      fontSize: '10px',
      fontFamily: 'primary',
      textAlign: 'center',
      textTransform: 'uppercase',
      fontWeight: 'bold',
      letterSpacing: '0.05em',
      px: 2,
      py: '2px',
      color: 'white',
      bg: 'primary',
      border: 0,
      borderRadius: isMedium ? '14px' : '8px',
      flexDirection: 'row',
      justifyContent: 'center',
      alignItems: 'center',
      padding: hasIcon ? '4px 8px' : '1px 5px',
      position: 'static',
      flex: 'none',
      order: 0,
      flexGrow: 0,
      margin: '0px 0px',
    }}
  />
)
