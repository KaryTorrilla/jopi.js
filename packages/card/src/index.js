import React from 'react'
import { Box } from '@oneloop/box'
import theme from '@oneloop/theme'

export const Card = ({ variant, active, ...props }) => {
  const variantValues = Object.values(theme.card)[Object.keys(theme.card).indexOf(variant)]
  const backgroundColor = active ? variantValues[':focus'].backgroundColor : variantValues.backgroundColorValue
  return (
    <Box sx={{ position: 'relative' }}>
      <Box
        tx='card'
        variant={variant}
        {...props}
        __css={{
          fontFamily: 'primary',
          display: 'flex',
          flexDirection: 'row',
          alignItems: 'center',
          backgroundColor: backgroundColor,
          position: 'static',
        }}
        />
    </Box>
  )
}
