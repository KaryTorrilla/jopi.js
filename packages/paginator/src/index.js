import React, { useState } from 'react'
import { Box } from '@oneloop/box'

export const Paginator = (props) => {
  const [state, setState] = useState(props)

  const handlePageChanged = (page) => {
    setState({
      currentPage: page,
      total: props.total,
      perPage: props.perPage,
    })
    // call request to refresh data
  }

  const lastPage = () => {
    const { total, perPage } = props
    return Math.max(Math.ceil(total / perPage), 1)
  }

  const makeUrlWindow = () => {
    // small slider
    if (lastPage() < 12) {
      return {
        first: _.range(1, lastPage() + 1),
        slider: null,
        last: null,
      }
    }

    // url slider
    const onBothSide = 6

    const { currentPage } = props

    // if slider too close to beginning
    if (currentPage <= onBothSide) {
      return {
        first: _.range(1, onBothSide + 3),
        slider: null,
        last: _.range(lastPage() - 1, lastPage() + 1),
      }
    }

    // if slider too close to ending
    if (currentPage > lastPage() - onBothSide) {
      return {
        first: _.range(1, 3),
        slider: null,
        last: _.range(lastPage() - (onBothSide + 2), lastPage() + 1),
      }
    }

    // get full slider
    return {
      first: _.range(1, 3),
      slider: _.range(currentPage - 3, currentPage + 4),
      last: _.range(lastPage() - 1, lastPage() + 1),
    }
  }

  const elements = () => {
    const urlWindow = makeUrlWindow()

    return _.compact([
      urlWindow.first,
      Array.isArray(urlWindow.slider) ? '...' : null,
      urlWindow.slider,
      Array.isArray(urlWindow.last) ? '...' : null,
      urlWindow.last,
    ])
  }

  const onFirstPage = () => {
    return props.currentPage <= 1
  }

  const hasMorePages = () => {
    return props.currentPage < lastPage()
  }

  const previousPage = () => {
    const { currentPage } = props
    return currentPage > 1 ? currentPage - 1 : 1
  }

  const nextPage = () => {
    const { currentPage } = props
    const lastPage = lastPage()
    return lastPage > currentPage ? currentPage + 1 : lastPage
  }

  const changePage = (page, e) => {
    e.preventDefault()
    //const { handlePageChanged } = props;
    handlePageChanged(page)
  }

  return (
    <Box sx={{ display: 'flex' }}>
      <Box
        sx={{
          paddingLeft: '0.5rem',
          paddingRight: '0.5rem',
        }}
      >
        {onFirstPage() ? (
          <span
            sx={{
              cursor: 'pointer',
              color: '#C8C9CA',
              fontStyle: 'normal',
              fontWeight: 'normal',
              fontSize: '16px',
              lineHeight: '22px',
            }}
          >
            {'<'}
          </span>
        ) : (
          <span
            sx={{
              cursor: 'pointer',
              color: '#C8C9CA',
              fontStyle: 'normal',
              fontWeight: 'normal',
              fontSize: '16px',
              lineHeight: '22px',
            }}
            onClick={(e) => changePage(previousPage(), e)}
          >
            {'<'}
          </span>
        )}
      </Box>
      {elements().map((element, index) => {
        const Element = () => {
          if (typeof element === 'string') {
            return (
              <Box
                sx={{
                  cursor: 'pointer',
                  color: '#C8C9CA',
                  fontStyle: 'normal',
                  fontWeight: 'normal',
                  fontSize: '16px',
                  lineHeight: '22px',
                  paddingLeft: '0.5rem',
                  paddingRight: '0.5rem',
                }}
              >
                {element}
              </Box>
            )
          }
          if (Array.isArray(element)) {
            return element.map((page) => {
              if (page === props.currentPage) {
                return (
                  <Box
                    sx={{
                      cursor: 'pointer',
                      color: '#D54936',
                      borderBottom: '2px solid #D54936',
                      paddingBottom: '3px',
                      fontStyle: 'normal',
                      fontWeight: 'normal',
                      fontSize: '16px',
                      lineHeight: '22px',
                      paddingLeft: '0.5rem',
                      paddingRight: '0.5rem',
                    }}
                    key={page}
                  >
                    {page}
                  </Box>
                )
              }
              return (
                <Box
                  sx={{
                    cursor: 'pointer',
                    color: '#C8C9CA',
                    fontStyle: 'normal',
                    fontWeight: 'normal',
                    fontSize: '16px',
                    lineHeight: '22px',
                    paddingLeft: '0.5rem',
                    paddingRight: '0.5rem',
                  }}
                  key={page}
                >
                  <span onClick={(e) => changePage(page, e)}>{page}</span>
                </Box>
              )
            })
          }
          return null
        }
        return <Element key={index} />
      })}
      <Box
        sx={{
          paddingLeft: '0.5rem',
          paddingRight: '0.5rem',
        }}
      >
        {hasMorePages() ? (
          <span
            sx={{
              cursor: 'pointer',
              color: '#C8C9CA',
              fontStyle: 'normal',
              fontWeight: 'normal',
              fontSize: '16px',
              lineHeight: '22px',
            }}
            onClick={(e) => changePage(nextPage(), e)}
          >
            {'>'}
          </span>
        ) : (
          <span
            sx={{
              cursor: 'pointer',
              color: '#C8C9CA',
              fontStyle: 'normal',
              fontWeight: 'normal',
              fontSize: '16px',
              lineHeight: '22px',
            }}
          >
            {'>'}
          </span>
        )}
      </Box>
    </Box>
  )
}
