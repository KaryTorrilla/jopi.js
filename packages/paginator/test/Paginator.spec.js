import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import { Paginator } from '../src'

describe('Paginator', () => {
  test('renders correctly 1', () => {
    const tree = renderer
      .create(
        <Paginator
          currentPage={1}
          total={1000}
          perPage={10}
          changePageNumber={() => console.log('change page')}
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('renders correctly 2', () => {
    const tree = renderer
      .create(
        <Paginator
          currentPage={20}
          total={150}
          perPage={25}
          changePageNumber={() => console.log('change page')}
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('renders correctly 3 - big numbers', () => {
    const tree = renderer
      .create(
        <Paginator
          currentPage={410}
          total={17000}
          perPage={37}
          changePageNumber={() => console.log('change page')}
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('Paginator with error in actualPage value 0', () => {
    const tree = renderer
      .create(
        <Paginator
          currentPage={0}
          total={1000}
          perPage={10}
          changePageNumber={() => console.log('change page')}
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('Paginator with error in currentPage value -20', () => {
    const tree = renderer
      .create(
        <Paginator
          currentPage={-20}
          total={1000}
          perPage={10}
          changePageNumber={() => console.log('change page')}
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('Paginator with error in currentPage value String', () => {
    const tree = renderer
      .create(
        <Paginator
          currentPage={'Hello world'}
          total={1000}
          perPage={10}
          changePageNumber={() => console.log('change page')}
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('Paginator with error, currentPage too big', () => {
    const tree = renderer
      .create(
        <Paginator
          currentPage={20}
          total={10}
          perPage={10}
          changePageNumber={() => console.log('change page')}
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('Paginator with error, perPage < 0', () => {
    const tree = renderer
      .create(
        <Paginator
          currentPage={20}
          total={100}
          perPage={-10}
          changePageNumber={() => console.log('change page')}
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('Paginator with total < perPage && (currentPage > 1) ', () => {
    const tree = renderer
      .create(
        <Paginator
          currentPage={2}
          total={23}
          perPage={30}
          changePageNumber={() => console.log('change page')}
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('Paginator with (total > perPage && total % perPage == 0) && ( currentPage <= total/perPage) ', () => {
    const tree = renderer
      .create(
        <Paginator
          currentPage={2}
          total={20}
          perPage={10}
          changePageNumber={() => console.log('change page')}
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('Paginator with (total > perPage && total % perPage == 0) && ( currentPage > total/perPage) ', () => {
    const tree = renderer
      .create(
        <Paginator
          currentPage={3}
          total={20}
          perPage={10}
          changePageNumber={() => console.log('change page')}
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('Paginator with (total > perPage && total % perPage != 0) && ( currentPage <= (1 + total / perPage)', () => {
    const tree = renderer
      .create(
        <Paginator
          currentPage={4}
          total={19}
          perPage={5}
          changePageNumber={() => console.log('change page')}
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('Paginator with (total > perPage && total % perPage != 0) && ( currentPage > (1 + total / perPage)', () => {
    const tree = renderer
      .create(
        <Paginator
          currentPage={5}
          total={19}
          perPage={5}
          changePageNumber={() => console.log('change page')}
        />
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
