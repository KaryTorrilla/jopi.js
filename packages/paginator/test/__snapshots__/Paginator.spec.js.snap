// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paginator Paginator with (total > perPage && total % perPage != 0) && ( currentPage <= (1 + total / perPage) 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.c2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  cursor: pointer;
  color: #C8C9CA;
  font-style: normal;
  font-weight: normal;
  font-size: 16px;
  line-height: 22px;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.c3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  cursor: pointer;
  color: #D54936;
  border-bottom: 2px solid #D54936;
  padding-bottom: 3px;
  font-style: normal;
  font-weight: normal;
  font-size: 16px;
  line-height: 22px;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <span
      onClick={[Function]}
      style={
        Object {
          "color": "#C8C9CA",
          "cursor": "pointer",
          "fontSize": "16px",
          "fontStyle": "normal",
          "fontWeight": "normal",
          "lineHeight": "22px",
        }
      }
    >
      &lt;
    </span>
  </div>
  <div
    className="c2"
  >
    <span
      onClick={[Function]}
    >
      1
    </span>
  </div>
  <div
    className="c2"
  >
    <span
      onClick={[Function]}
    >
      2
    </span>
  </div>
  <div
    className="c2"
  >
    <span
      onClick={[Function]}
    >
      3
    </span>
  </div>
  <div
    className="c3"
  >
    4
  </div>
  <div
    className="c1"
  />
</div>
`;

exports[`Paginator Paginator with (total > perPage && total % perPage != 0) && ( currentPage > (1 + total / perPage) 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

<div
  className="c0"
>
  &lt;
   
  &gt;
   
</div>
`;

exports[`Paginator Paginator with (total > perPage && total % perPage == 0) && ( currentPage <= total/perPage)  1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.c2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  cursor: pointer;
  color: #C8C9CA;
  font-style: normal;
  font-weight: normal;
  font-size: 16px;
  line-height: 22px;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.c3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  cursor: pointer;
  color: #D54936;
  border-bottom: 2px solid #D54936;
  padding-bottom: 3px;
  font-style: normal;
  font-weight: normal;
  font-size: 16px;
  line-height: 22px;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <span
      onClick={[Function]}
      style={
        Object {
          "color": "#C8C9CA",
          "cursor": "pointer",
          "fontSize": "16px",
          "fontStyle": "normal",
          "fontWeight": "normal",
          "lineHeight": "22px",
        }
      }
    >
      &lt;
    </span>
  </div>
  <div
    className="c2"
  >
    <span
      onClick={[Function]}
    >
      1
    </span>
  </div>
  <div
    className="c3"
  >
    2
  </div>
  <div
    className="c1"
  />
</div>
`;

exports[`Paginator Paginator with (total > perPage && total % perPage == 0) && ( currentPage > total/perPage)  1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

<div
  className="c0"
>
  &lt;
   
  &gt;
   
</div>
`;

exports[`Paginator Paginator with error in actualPage value 0 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

<div
  className="c0"
>
  &lt;
   
  &gt;
   
</div>
`;

exports[`Paginator Paginator with error in currentPage value -20 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

<div
  className="c0"
>
  &lt;
   
  &gt;
   
</div>
`;

exports[`Paginator Paginator with error in currentPage value String 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.c2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  cursor: pointer;
  color: #C8C9CA;
  font-style: normal;
  font-weight: normal;
  font-size: 16px;
  line-height: 22px;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <span
      onClick={[Function]}
      style={
        Object {
          "color": "#C8C9CA",
          "cursor": "pointer",
          "fontSize": "16px",
          "fontStyle": "normal",
          "fontWeight": "normal",
          "lineHeight": "22px",
        }
      }
    >
      &lt;
    </span>
  </div>
  <div
    className="c2"
  >
    <span
      onClick={[Function]}
    >
      1
    </span>
  </div>
  <div
    className="c2"
  >
    <span
      onClick={[Function]}
    >
      2
    </span>
  </div>
  <div
    className="c2"
  >
    ...
  </div>
  <div
    className="c2"
  >
    ...
  </div>
  <div
    className="c2"
  >
    <span
      onClick={[Function]}
    >
      99
    </span>
  </div>
  <div
    className="c2"
  >
    <span
      onClick={[Function]}
    >
      100
    </span>
  </div>
  <div
    className="c1"
  />
</div>
`;

exports[`Paginator Paginator with error, currentPage too big 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.c2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  cursor: pointer;
  color: #C8C9CA;
  font-style: normal;
  font-weight: normal;
  font-size: 16px;
  line-height: 22px;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <span
      onClick={[Function]}
      style={
        Object {
          "color": "#C8C9CA",
          "cursor": "pointer",
          "fontSize": "16px",
          "fontStyle": "normal",
          "fontWeight": "normal",
          "lineHeight": "22px",
        }
      }
    >
      &lt;
    </span>
  </div>
  <div
    className="c2"
  >
    <span
      onClick={[Function]}
    >
      1
    </span>
  </div>
  <div
    className="c1"
  />
</div>
`;

exports[`Paginator Paginator with error, perPage < 0 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

<div
  className="c0"
>
  &lt;
   
  &gt;
   
</div>
`;

exports[`Paginator Paginator with total < perPage && (currentPage > 1)  1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

<div
  className="c0"
>
  &lt;
   
  &gt;
   
</div>
`;

exports[`Paginator renders correctly 1 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.c2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  cursor: pointer;
  color: #D54936;
  border-bottom: 2px solid #D54936;
  padding-bottom: 3px;
  font-style: normal;
  font-weight: normal;
  font-size: 16px;
  line-height: 22px;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.c3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  cursor: pointer;
  color: #C8C9CA;
  font-style: normal;
  font-weight: normal;
  font-size: 16px;
  line-height: 22px;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
  <div
    className="c2"
  >
    1
  </div>
  <div
    className="c3"
  >
    <span
      onClick={[Function]}
    >
      2
    </span>
  </div>
  <div
    className="c3"
  >
    <span
      onClick={[Function]}
    >
      3
    </span>
  </div>
  <div
    className="c3"
  >
    <span
      onClick={[Function]}
    >
      4
    </span>
  </div>
  <div
    className="c3"
  >
    <span
      onClick={[Function]}
    >
      5
    </span>
  </div>
  <div
    className="c3"
  >
    <span
      onClick={[Function]}
    >
      6
    </span>
  </div>
  <div
    className="c3"
  >
    <span
      onClick={[Function]}
    >
      7
    </span>
  </div>
  <div
    className="c3"
  >
    <span
      onClick={[Function]}
    >
      8
    </span>
  </div>
  <div
    className="c3"
  >
    ...
  </div>
  <div
    className="c3"
  >
    <span
      onClick={[Function]}
    >
      99
    </span>
  </div>
  <div
    className="c3"
  >
    <span
      onClick={[Function]}
    >
      100
    </span>
  </div>
  <div
    className="c1"
  >
    <span
      onClick={[Function]}
      style={
        Object {
          "color": "#C8C9CA",
          "cursor": "pointer",
          "fontSize": "16px",
          "fontStyle": "normal",
          "fontWeight": "normal",
          "lineHeight": "22px",
        }
      }
    >
      &gt;
    </span>
  </div>
</div>
`;

exports[`Paginator renders correctly 2 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

<div
  className="c0"
>
  &lt;
   
  &gt;
   
</div>
`;

exports[`Paginator renders correctly 3 - big numbers 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.c2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  cursor: pointer;
  color: #C8C9CA;
  font-style: normal;
  font-weight: normal;
  font-size: 16px;
  line-height: 22px;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

.c3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  cursor: pointer;
  color: #D54936;
  border-bottom: 2px solid #D54936;
  padding-bottom: 3px;
  font-style: normal;
  font-weight: normal;
  font-size: 16px;
  line-height: 22px;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <span
      onClick={[Function]}
      style={
        Object {
          "color": "#C8C9CA",
          "cursor": "pointer",
          "fontSize": "16px",
          "fontStyle": "normal",
          "fontWeight": "normal",
          "lineHeight": "22px",
        }
      }
    >
      &lt;
    </span>
  </div>
  <div
    className="c2"
  >
    <span
      onClick={[Function]}
    >
      1
    </span>
  </div>
  <div
    className="c2"
  >
    <span
      onClick={[Function]}
    >
      2
    </span>
  </div>
  <div
    className="c2"
  >
    ...
  </div>
  <div
    className="c2"
  >
    <span
      onClick={[Function]}
    >
      407
    </span>
  </div>
  <div
    className="c2"
  >
    <span
      onClick={[Function]}
    >
      408
    </span>
  </div>
  <div
    className="c2"
  >
    <span
      onClick={[Function]}
    >
      409
    </span>
  </div>
  <div
    className="c3"
  >
    410
  </div>
  <div
    className="c2"
  >
    <span
      onClick={[Function]}
    >
      411
    </span>
  </div>
  <div
    className="c2"
  >
    <span
      onClick={[Function]}
    >
      412
    </span>
  </div>
  <div
    className="c2"
  >
    <span
      onClick={[Function]}
    >
      413
    </span>
  </div>
  <div
    className="c2"
  >
    ...
  </div>
  <div
    className="c2"
  >
    <span
      onClick={[Function]}
    >
      459
    </span>
  </div>
  <div
    className="c2"
  >
    <span
      onClick={[Function]}
    >
      460
    </span>
  </div>
  <div
    className="c1"
  >
    <span
      onClick={[Function]}
      style={
        Object {
          "color": "#C8C9CA",
          "cursor": "pointer",
          "fontSize": "16px",
          "fontStyle": "normal",
          "fontWeight": "normal",
          "lineHeight": "22px",
        }
      }
    >
      &gt;
    </span>
  </div>
</div>
`;
